commonmark-ext-styles/src/main/java/org/commonmark/ext/styles/internal/StylesDelimiterProcessor.java

```diff
@@ -0,0 +1,51 @@
+package org.commonmark.ext.styles.internal;
+
+import org.commonmark.ext.styles.Styles;
+import org.commonmark.node.Node;
+import org.commonmark.node.Text;
+import org.commonmark.parser.delimiter.DelimiterProcessor;
+import org.commonmark.parser.delimiter.DelimiterRun;
+
+public class StylesDelimiterProcessor implements DelimiterProcessor {
+
+    @Override
+    public char getOpeningCharacter() {
+        return '{';
+    }
+
+    @Override
+    public char getClosingCharacter() {
+        return '}';
+    }
+
+    @Override
+    public int getMinLength() {
+        return 1;
+    }
+
+    @Override
+    public int getDelimiterUse(DelimiterRun opener, DelimiterRun closer) {
+        return 1;
+    }
+
+    @Override
+    public void process(Text opener, Text closer, int delimiterCount) {
+        StringBuilder styleString = new StringBuilder();
+        Node tmp = opener.getNext();
+        while (tmp != null && tmp != closer) {
+            Node next = tmp.getNext();
+            if (tmp instanceof Text) {
```

Comment:
Pandoc's behavior is to fall back to text if it can't be parsed, e.g.:
* https://pandoc.org/try/?text=!%5B%5D(file.jpg)%7Bwidth%3D50+*test*%7D&from=markdown&to=&standalone=0
* https://pandoc.org/try/?text=!%5B%5D(file.jpg)%7Bwidth%3D50+baa%7D&from=markdown&to=&standalone=0

Can you implement that here (and add a test?). We'd need to do some more parsing here and bail for other node types.

Reply:
Added support for falling back to text, and added/updated tests to confirm this behaviour.